test/network_generation/test_logger.rb

```diff
@@ -115,6 +133,46 @@
             refute @dataflow_graph.has_edge?(task, logger)
         end
 
+        it "cleans up connections while sharing a logger across deployments" do
+            deployment2 = syskit_stub_deployment("deployment2", deployment_m)
+            task2 = deployment2.task "task"
+            syskit_engine.should_receive(:deployment_tasks)
+                         .and_return([@deployment, deployment2])
+            syskit_engine.should_receive(:deployed_tasks)
+                         .and_return([@task, task2])
+
+            logger = deployment.task "deployment_Logger"
+            deployment2.logger_task = logger
+
+            Syskit::NetworkGeneration::LoggerConfigurationSupport
+                .add_logging_to_network(syskit_engine, plan)
+
+            flexmock(deployment).should_receive(log_port?: false)
+            flexmock(deployment2).should_receive(log_port?: false)
+            Syskit::NetworkGeneration::LoggerConfigurationSupport
+                .add_logging_to_network(syskit_engine, plan)
```

Comment:
why do you have to call `#add_logging_to_network` twice?

Reply:
I'm essentially using the first one to setup connections, that are then cleaned up by the second.